src/main/java/codeu/controller/ProfilePageServlet.java

```diff
@@ -71,5 +76,33 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
        * Send AboutMe, and Interests data to UserProfile object
        * Update UserProfileStore with new UserProfile object. 
        */
+      String username = (String) request.getSession().getAttribute("user");
+      if (!userStore.isUserRegistered(username)) {
+        request.setAttribute("error", "User is not registered");
+        request.getRequestDispatcher("/WEB-INF/view/profilepage.jsp").forward(request, response);
+        return;
+      }
+      User user = userStore.getUser(username);
+      UUID userId = userStore.getUser(username).getId();
+      UserProfile userProfile = null;
+      String aboutMe = request.getParameter("aboutMe");
+      String profilePicture = request.getParameter("profilePicture");
+      String category = request.getParameter("myFavorites");
+      String subCategory = request.getParameter("subcategory");
+
+      if (userProfileStore.isUserProfileCreated(userId)) {
+        userProfile = userProfileStore.getUserProfile(userId);
+        userProfile.setLastTimeOnline(Instant.now());
```

Comment:
About setting the last time online (for future reference I guess), I wonder how often we should update it or how we can go about this. When we do the log out function, we definitely want to update it then

Reply:
+1, this is a great thing to think about! It seems to me like setting this when the user logs in or out is a great basic version to get working first.

Reply:
If we're updating when they login, that should be easy to do, however for when they logout, i think i'd have to figure out a way to track when the user leaves the domain

Most people don't formally log out to sites, they just tend to leave the site in general.